webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -0,0 +1,55 @@
+import React, { createContext } from 'react';
+import { Modal } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import ForemanModalHeader from './subcomponents/ForemanModalHeader';
+import ForemanModalFooter from './subcomponents/ForemanModalFooter';
+import { extractModalNodes } from './helpers';
+
+export const ModalContext = createContext(null);
+
+const ForemanModal = props => {
+  const { isOpen, onClose, title, ...propsToPassDown } = props; // don't pass down those two as it causes PF problems
+  // Extract header and footer from children, if provided
+  const { headerChild, footerChild, otherChildren } = extractModalNodes(
+    props.children
+  );
+  const context = {
+    isOpen,
+    onClose,
+    title,
+  };
+
+  return (
+    <ModalContext.Provider value={context}>
+      {/* Change the name of the props we are passing down to Modal to conform to Patternfly 3 api */}
+      <Modal
+        onHide={onClose}
+        show={isOpen}
+        className="foreman-modal"
+        {...propsToPassDown}
+      >
+        {headerChild}
+        <Modal.Body>{otherChildren}</Modal.Body>
+        {footerChild}
+      </Modal>
+    </ModalContext.Provider>
+  );
+};
+
+// Header and Footer use the provided children, or default markup if none provided
+
+ForemanModal.Header = ForemanModalHeader;
+ForemanModal.Footer = ForemanModalFooter;
```

Comment:
would it make sense adding `ForemanModal.Context`?

Reply:
@sharvit I was excited for a second and tried it, but turns out it caused lots of problems and I ended up changing it back.